vendor/github.com/mutecomm/go-sqlcipher/aes.c

```diff
@@ -34,7 +18,12 @@
 
 #ifdef LTC_RIJNDAEL
 
-#ifndef ENCRYPT_ONLY 
+#include "mbtls_aes.h"
+
+static mbedtls_aes_context ctx_encrypt;
+
+#ifndef ENCRYPT_ONLY
+static mbedtls_aes_context ctx_decrypt;
```

Comment:
Using thread local variables seems to fix the DB corruption (hopefully the x86_64 crash also)

```suggestion
static __thread mbedtls_aes_context ctx_encrypt;

#ifndef ENCRYPT_ONLY
static __thread mbedtls_aes_context ctx_decrypt;
```

Reply:
Will give it a try!

Reply:
works \o/